python/rawdatautils/unpack/dataclasses.py

```diff
@@ -101,10 +101,13 @@ class TriggerRecordData(RecordDataBase):
     max_sequence_number: int
     total_size_bytes: int
     trigger_time : datetime = field(init=False)
+    trigger_type_bits: list[int] = field(init=False)
+    trigger_type_names: list[str] = field(init=False)
 
     def __post_init__(self):
         self.trigger_time = dts_to_datetime(self.trigger_timestamp_dts)
-
+        self.trigger_type_bits = [ trgdataformats.TriggerCandidateData.Type(i) for i in range(64) if (self.trigger_type & (1<<i))!=0 ]
+        self.trigger_type_names = [ str(trgdataformats.TriggerCandidateData.Type(i))[6:] for i in range(64) if (self.trigger_type & (1<<i))!=0 ]
```

Comment:
It would be better to use the `trgdataformats.TriggerCandidateData.Type(i).value` and `*.name`. Otherwise, `trigger_type_bits` will actually be `list[trgdataformats.TriggerCandidateData.Type]` and not `list[int]`. Using `*.name` would return just the name with the `k` prefix, e.g. `kTiming`, so getting the substring wouldn't be necessary.
```suggestion
        self.trigger_type_bits = [ trgdataformats.TriggerCandidateData.Type(i).value for i in range(64) if (self.trigger_type & (1<<i))!=0 ]
        self.trigger_type_names = [ trgdataformats.TriggerCandidateData.Type(i).name for i in range(64) if (self.trigger_type & (1<<i))!=0 ]
```